arroyo-console/src/routes/connections/ConfigureProfile.tsx

```diff
@@ -0,0 +1,232 @@
+import { JSONSchema7 } from 'json-schema';
+import {
+  Button,
+  HStack,
+  Heading,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Spacer,
+  Stack,
+  Text,
+  useDisclosure,
+  IconButton,
+  Box,
+  LinkBox,
+  LinkOverlay,
+  useToast,
+} from '@chakra-ui/react';
+import { useState } from 'react';
+import { JsonForm } from './JsonForm';
+import { AddIcon, ArrowBackIcon, DeleteIcon, InfoOutlineIcon } from '@chakra-ui/icons';
+import { ConnectionProfile, Connector, del, useConnectionProfiles } from '../../lib/data_fetching';
+import { CreateConnectionState } from './CreateConnection';
+import { formatError } from '../../lib/util';
+import { CreateProfile } from './CreateProfile';
+
+function ClusterViewerModal({
+  isOpen,
+  onClose,
+  profile,
+  connector,
+  schema,
+}: {
+  isOpen: boolean;
+  onClose: () => void;
+  profile: ConnectionProfile;
+  connector: string;
+  schema: JSONSchema7;
+}) {
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} size={'2xl'}>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>Viewing {connector} connection</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody p={4} px={7}>
+          <JsonForm
+            schema={schema}
+            hasName={true}
+            error={null}
+            initial={profile.config}
+            button={'Close'}
+            buttonColor={'gray'}
+            readonly={true}
+            onSubmit={async () => onClose()}
+          />
+        </ModalBody>
+      </ModalContent>
+    </Modal>
+  );
+}
+
+const ClusterChooser = ({
+  connector,
+  connections,
+  onSubmit,
+  schema,
+}: {
+  connector: string;
+  connections: Array<ConnectionProfile>;
+  onSubmit: (c: string) => void;
+  schema: JSONSchema7;
+}) => {
+  const [editingProfile, setEditingProfile] = useState<any>({});
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const toast = useToast();
+  const { mutateConnectionProfiles } = useConnectionProfiles();
+
+  const deleteProfile = async (profile: ConnectionProfile) => {
+    const { error } = await del('/v1/connection_profiles/{id}', {
+      params: { path: { id: profile.id } },
+    });
+    mutateConnectionProfiles();
+    if (error) {
+      toast({
+        title: 'Failed to delete connection profile',
+        description: formatError(error),
+        status: 'error',
+        duration: 9000,
+        isClosable: true,
+      });
+    } else {
+      toast({
+        title: 'Connection profile deleted',
+        description: `Successfully deleted connection profile ${profile.name}`,
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+      });
+    }
+  };
+
+  return (
```

Comment:
I think it's a good practice to break things down into variables so that the return statements are relatively small, so you can get an idea of the overall layout/content by looking at it.

Reply:
Hmm, it's not clear to me how much clearer that would make this, since it would just be a couple of variable declarations and then returning them. In the end, it's the same amount of code that needs to be on screen to understand what's happening.